### ISSUE TITLE
[display] @type docs on types

### ISSUE THREAD

**Issue by Gama11** [HaxeFoundation/haxe#5709]
abdc4de added this for fields,

![](http://i.imgur.com/TWW4u9f.png)

but it's still missing on types:

![](http://i.imgur.com/G4siGA5.png)

``` haxe
/** doc **/
class Main {
    /** doc **/
    public static function main() {}
}
```

**Comment by Simn** [comment]
To clarify, we want this on explicit occurrences of the type, but not when something merely has that type (like `"foo"` for String), right?

**Comment by Gama11** [comment]
Yeah, I think having it for "implicit occurences" would be a bit overkill.

**Comment by Gama11** [comment]
Nice! Doesn't work for constructor calls yet though:

![](http://i.imgur.com/oL9JZW3.png)

**Comment by Gama11** [comment]
Also, should this work for basic types like `String`?

**Comment by Simn** [comment]
Doesn't work for class relations either. It's a bit of a problem because we have to distinguish the explicit from the implicit cases (internally both just load a type instance).

It doesn't work for String and other `@:coreApi` classes at the moment because that requires doc merging, which requires a typer context, which we don't have at the place we're dealing with this. I thought about simply creating a new one because it really shouldn't matter.

**Comment by Gama11** [comment]
This still seems to fail in some (seemingly) very simple cases, unrelated to `@:coreApi` etc:

![](https://i.imgur.com/iUYAJXk.png)

**Comment by Simn** [comment]
Should be better now. Keeping this open so we add a test.

**Comment by Gama11** [comment]
Nice, that's much better already. :) Still an issue with core classes like `String` though.

**Comment by Gama11** [comment]
>Still an issue with core classes like `String` though.

This doesn't seem to reproduce anymore.
